SHARED-CI.md

```diff
@@ -154,3 +154,71 @@ to avoid those restrictions.
 [trusted-artifacts generator]: https://github.com/konflux-ci/build-definitions/tree/main/task-generator/trusted-artifacts
 [GITHUB_TOKEN]: https://docs.github.com/en/actions/concepts/security/github_token
 [tekton-catalog-structure]: https://github.com/tektoncd/catalog?tab=readme-ov-file#catalog-structure
+
+### Task Integration Tests
+
+- workflow: [`.github/workflows/run-task-tests.yaml`](.github/workflows/run-task-tests.yaml)
+- script: [`.github/scripts/test_tekton_tasks.sh`](.github/scripts/test_tekton_tasks.sh)
+
+This workflow automatically runs integration tests for any Tekton Task that is changed in a pull request. It spins up a temporary Kubernetes (Kind) cluster, deploys Tekton, and then executes the tests defined for the modified task.
+
+#### How to Add a Test
+
+1. Create a `tests` directory inside the task's versioned folder.  
+
+2. Inside the `tests` directory, create a test file named `test-*.yaml` (for example, `test-hello.yaml`).  
+   - The script **automatically** discovers tests based on this naming convention.  
+
+3. The file must define a Tekton `kind: Pipeline` object.  
+
+4. The Pipeline must declare a workspace named exactly `tests-workspace`.  
+   - The test script will **automatically** provide storage for this workspace when it runs the pipeline.  
+
+5. Optionally, add a `pre-apply-task-hook.sh` to the `tests` directory.
+
+#### Example Structure
+
+```plaintext
+task
+└── hello
+    └── 0.1
+        ├── hello.yaml
+        └── tests                         👈 Test directory
+            └── test-hello.yaml           👈 A Pipeline named test-*.yaml
+            └── pre-apply-task-hook.sh    👈 Optional hook
+```
+
+#### Using a `pre-apply-task-hook.sh`
+
+In some cases, your Task may require certain Kubernetes resources, like **Secrets** or **ConfigMaps**, to exist in the namespace before the Task itself is applied to the cluster.
+
+To handle this, you can create an optional shell script named `pre-apply-task-hook.sh` and place it inside the `tests` directory.
+
+If this script exists, the test runner will execute it **after creating the test namespace but before applying the task**. The script receives two arguments:
+
+- `$1`: The path to a temporary copy of the task's YAML file.  
```

Comment:
It may not be clear why this is useful. We could mention that the pre-apply-hook can modify the task YAML (for example, to reduce/remove the resource requests and limits) or do this in the example hook script below